### PR TITLE
fix(flat-table-header): alternative color not changing according theme

### PR DESCRIPTION
**Pull request by AlexKryzh**
Closes: FE-5319

### Proposed behaviour

Fix for the "Light" theme of the Flat Table header alternative background color then attribute trigger alternativeBgColor is used: https://github.com/Sage/carbon/issues/5136 

The theme "dark" is already using correct colors, themes "transparent-base" and "transparent-white" don't has alternative background colors, so they are using the same Flat Table header colors.

image

Sandbox with bug: https://codesandbox.io/s/modest-shaw-n4hxlv

### Current behaviour

Flat Table header with alternative background color is showing wrong colors, used in "dark" theme.
image

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [ ] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

Sandbox with bug: https://codesandbox.io/s/modest-shaw-n4hxlv

### CODE REVIEW

**Review by DipperTheDan — COMMENTED** [review]
Hi Alex, Thanks for the PR. Two things need addressing firstly though.

1. The commit message needs to be in a format of `fix(flat-table-header): commit message here...`

2. Your PR seems to have introduced a visual bug as demonstrated in the screenshot below. the flat-table-header looks like it has a thicker border left than what it should have.
Screenshot 2022-08-19 at 09 09 45

**Comment by AlexKryzh** [comment]
> Hi @AlexKryzh, thanks for the PR. Two things need addressing firstly though.
> 
> 1. The commit message needs to be in a format of `fix(flat-table-header): commit message here...`
> 2. Your PR seems to have introduced a visual bug as demonstrated in the screenshot below. the flat-table-header looks like it has a thicker border left than what it should have.
>    Screenshot 2022-08-19 at 09 09 45

Ok, I made required changes.

image

File: src/components/flat-table/flat-table-header/flat-table-header.component.js [hunk 1]
```diff
@@ -23,6 +24,7 @@ const FlatTableHeader = ({
   ...rest
 }) => {
   const ref = useRef(null);
+  const themeContext = useContext(FlatTableThemeContext);
```
Comment (edleeks87) [inline src/components/flat-table/flat-table-header/flat-table-header.component.js:27]:
```suggestion
  const { colorTheme } = useContext(FlatTableThemeContext);
```

File: src/components/flat-table/flat-table-header/flat-table-header.component.js [hunk 2]
```diff
@@ -36,6 +38,7 @@ const FlatTableHeader = ({
       leftPosition={leftPosition || 0}
       makeCellSticky={!!reportCellWidth}
       align={align}
+      colorTheme={themeContext.colorTheme}
```
Comment (edleeks87) [inline src/components/flat-table/flat-table-header/flat-table-header.component.js:41]:
```suggestion
      colorTheme={colorTheme}
```

File: src/components/flat-table/flat-table-header/flat-table-header.component.js (hunk 1 above)
Comment (AlexKryzh) [reply src/components/flat-table/flat-table-header/flat-table-header.component.js]: ok, changed.

File: src/components/flat-table/flat-table-header/flat-table-header.component.js (hunk 2 above)
Comment (AlexKryzh) [reply src/components/flat-table/flat-table-header/flat-table-header.component.js]: ok, changed.

**Review by edleeks87 — APPROVED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Pull request merged**